examples/pytorch/ogb/ogbn-mag/hetero_rgcn.py

```diff
@@ -383,27 +394,49 @@ def main(args):
 
     for run in range(args.runs):
 
-        embed_layer.reset_parameters()
-        model.reset_parameters()
+        try:
+            embed_layer.reset_parameters()
+            model.reset_parameters()
+        except:
+            # old pytorch version doesn't support reset_parameters() API
+            pass
 
         # optimizer
         all_params = itertools.chain(
             model.parameters(), embed_layer.parameters()
         )
         optimizer = th.optim.Adam(all_params, lr=0.01)
 
-        logger = train(
-            g,
-            model,
-            embed_layer,
-            optimizer,
-            train_loader,
-            split_idx,
-            labels,
-            logger,
-            device,
-            run,
-        )
+        if args.num_workers != 0 and device == "cpu" and is_support_affinity(v_t):
+            expected_max = int(psutil.cpu_count(logical=False)/2)
```

Comment:
I think the physical core count should be `psutil.cpu_count(logical=False)`. Can you double-check?
Also, below if condition (L412) should be `args.num_workers >= expected_max:`, otherwise it will crash.

Reply:
@chang-l , yes, I checked before and that was also why I thought ld_cores/computer_cores are necessary,
I have nproc as 96, and psutil.cpu_count(logical=False) as 48, but if I set num_workers as 48 while leave `enable_cpu_affinity()` empty with args, it go crashing. And set num_workers as 24, it is running well

Reply:
Thanks @xuechendi . Sorry for the late reply. 
I think it crashed because of this line?https://github.com/dmlc/dgl/blob/b13092172db668c26da487e8f782c7cf1caa92a6/python/dgl/dataloading/dataloader.py#L930-L931
which means your num_workers should be _strictly_ less than physical cores (not equal). Can you please try setting num_workers to 47? That's also why I said: 
>  below if condition (L412) should be args.num_workers >= expected_max:

Reply:
@chang-l , I see, I fixed the expected num_workers check in codes, and verified with num_workers as 47, appeared to be runnable
![image](https://user-images.githubusercontent.com/4355494/196254018-847177b3-1a9f-44ec-bd12-9407ad63dae2.png)